### PR TITLE
ci: github action to automate sanity suite test on browserstack

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

A new GitHub action to trigger sanity suites on Browserstack.

## Linear task (optional)

[Linear task link](https://linear.app/rudderstack/issue/SDK-2195/automate-js-sdk-sanity-test-on-browserstack)

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new workflow for running sanity tests on BrowserStack, allowing users to manually initiate testing on demand with options for environment selection and SDK version.

### CODE REVIEW

**Review by saikumarrs — CHANGES_REQUESTED** [review]
There is no need for a separate action to trigger the sanity test. It should be automatically triggered when staging and production SDKs are released.

**Comment by MoumitaM** [comment]
> There is no need for a separate action to trigger the sanity test. It should be automatically triggered when staging and production SDKs are released.

This will be addressed at later stages. Not in this PR.

**Review by saikumarrs — DISMISSED** [review]

File: .github/workflows/browserstack-sanity.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Run sanity suite on BrowserStack
+
+on:
+  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .github/workflows/browserstack-sanity.yml:4]: Done

File: .github/workflows/browserstack-sanity.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          - production
+      sdkVersion:
+        description: Version of the SDK
+        required: true
+
+jobs:
+  trigger:
+    runs-on: [self-hosted, Linux, X64]
+    steps:
+      - name: Trigger target repo workflow
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .github/workflows/browserstack-sanity.yml]: done

**Review by saikumarrs — APPROVED** [review]

File: .github/workflows/browserstack-sanity.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          - production
+      sdkVersion:
+        description: Version of the SDK
+        required: true
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+        default: production
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply .github/workflows/browserstack-sanity.yml:20]: not required

**Pull request merged**